src/server/game/Conditions/ConditionMgr.h

```diff
@@ -172,7 +172,8 @@ enum ConditionSourceType
     CONDITION_SOURCE_TYPE_AREATRIGGER                    = 28,
     CONDITION_SOURCE_TYPE_CONVERSATION_LINE              = 29,
     CONDITION_SOURCE_TYPE_AREATRIGGER_CLIENT_TRIGGERED   = 30,
-    CONDITION_SOURCE_TYPE_MAX                            = 31  // MAX
+    CONDITION_SOURCE_TYPE_TRAINER                        = 31,
```

Comment:
Name is CONDITION_SOURCE_TYPE_TRAINER_SPELL (you are filtering individual spells, not access to trainer)

Reply:
Updated. I've also changed the name of the container